File: cpp/830_较大分组的位置_easy.cpp
class Solution {
public:
    vector<vector<int>> largeGroupPositions(string s) {
        int N = s.length();
        if (N < 3)
            return {};
        int i=0;
        vector<vector<int>> ans;
        while (i+2 < N){
            if (s[i] == s[i+1] && s[i+1] == s[i+2]){
                int start = i;
                i += 2;
                while (i+1<N && s[i]==s[i+1])
                    i++;
                int end = i;
                ans.push_back({start, end});
            }
            i++;
        }
        return ans;
    }
};

File: python/830_较大分组的位置_easy.py
class Solution:
    def largeGroupPositions(self, s: str) -> List[List[int]]:
        N = len(s)
        if N < 3:
            return []
        ans = []
        i = 0
        while i + 2 < N:
            if s[i] == s[i+1] and s[i+1] == s[i+2]:
                start = i
                i += 2
                while(i+1<N and s[i]==s[i+1]):
                    i += 1
                end = i
                ans.append([start, end])
            i += 1
        return ans